genesiscommunitysuccess/docs
# chore: convert sync plugin to use typescript FUI-1754

This change doesn't alter any of the documentation files themselves. Rather, it converts the existing sync tool plugin to use typescript. No features have been added, so the current functionality of the tool should be exactly the same. 

* This has been turned into a private npm package rather than just a bunch of files
* Package converted to typescript
* The `processedMap` file remains javascript so you don't need to run any `tsc` transpiles when just bumping and syncing versions
* Plugin readme updated
* I decided to check in the javascript files from the plugin into source control. This means that nobody has to worry about any extra build steps outside of the few who are actually working on changing this plugin.
* Following the previous point, I added a `.gitattributes` to hide the diffs of the js files as there is no point revieiwing them

Testing steps
```
1. Check out this branch
2. `npm i` to run the install, linking the new package to the main learn repo
3. `npm run start:copy_docs`
4. The app should boot and no changes in git
5. Open `plugins/api-docs/processedMap.js` and comment out the `foudnation-forms` line
6. `npm run start:copy_docs`
```
The app should still boot, and the sync tool should be trying to change `foundation-forms` readme back to match FUI. Once you've tested you're seeing the same you can reset these changes, as we don't want to override the forms files.
